src/main/java/ui/UpdateProgressWindow.java

```diff
@@ -0,0 +1,139 @@
+package ui;
+
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressBar;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import updater.LabeledDownloadProgressBar;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * UpdateProgressWindow is the class that will display update download progresses
+ */
+public class UpdateProgressWindow {
+    private static final String WINDOW_TITLE = "Update Progress";
+    private static final String LABEL_NO_DOWNLOAD = "There is no update being downloaded.";
+
+    private Stage window;
+    private VBox windowMainLayout;
+    private final List<LabeledDownloadProgressBar> downloads;
+
+    public UpdateProgressWindow() {
+        this.downloads = new ArrayList<>();
+        this.initProgressWindow();
+    }
+
+    /**
+     * shows update download progress window
+     */
+    public void showWindow() {
+        Platform.runLater(() -> {
+                    window.show();
+                    window.toFront();
+                }
+        );
+    }
+
+    /**
+     * hides update download progress window
+     */
+    public void hideWindow() {
+        Platform.runLater(window::hide);
+    }
+
+    /**
+     * Initialize the progress window
+     */
+    private void initProgressWindow() {
+        Stage stage = new Stage();
+        stage.setTitle(WINDOW_TITLE);
+
+        windowMainLayout = new VBox();
+
+        Group root = new Group();
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+
+        scene.setRoot(windowMainLayout);
+
+        window = stage;
+
+        reloadWindowLayout();
+    }
+
+    /**
+     * Create a new download progress bar
+     *
+     * @param downloadUrl url of download as identifier to the progress bar
+     * @param downloadLabel label of progress bar
+     * @return the method to set the progress to the progress bar of the download progress tracker
+     */
+    public Consumer<Double> createNewDownloadProgressBar(URL downloadUrl, String downloadLabel) {
```

Comment:
This method is doing two things. Maybe we could just have this method return the progress bar, and pass the consumer at the call site instead?